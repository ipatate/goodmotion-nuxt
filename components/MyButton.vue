<template>
  <nuxt-link
    :to="this.to"
    class="font-bold py-2 px-4 text-white"
    :class="btnClass"
  >
    <slot />
  </nuxt-link>
</template>

<script>
export default {
  props: {
    to: {
      type: String,
      default: null,
    },
    round: {
      type: Boolean,
      default: false,
    },
    round: {
      type: Boolean,
      default: false,
    },
    type: {
      type: String,
      default: 'primary',
    },
    overclass: {
      type: String,
      default: '',
    },
  },
  computed: {
    btnClass() {
      const classes = [];
      if (this.round) {
        classes.push('rounded-full');
      }
      if (this.type === 'primary') {
        classes.push('bg-blue-500 hover:bg-blue-700');
      } else if (this.type === 'secondary') {
        classes.push('bg-gray-500 hover:bg-gray-700');
      }
      return `${classes.join(' ')} ${this.overclass}`;
    },
  },
};
</script>
