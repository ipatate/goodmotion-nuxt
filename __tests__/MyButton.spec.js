import {mount} from '@vue/test-utils';
import MyButton from '../components/MyButton.vue';

describe('Mounted App', () => {
  const wrapper = mount(MyButton, {
    stubs: {
      NuxtLink: true,
    },
  });

  test('snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
