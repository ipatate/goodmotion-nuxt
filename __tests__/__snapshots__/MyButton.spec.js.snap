// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted App snapshot 1`] = `<nuxtlink-stub class="font-bold py-2 px-4 text-white bg-blue-500 hover:bg-blue-700 "></nuxtlink-stub>`;
